compiler/rustc_lint/Cargo.toml

```diff
@@ -5,6 +5,7 @@ version = "0.0.0"
 edition = "2018"
 
 [dependencies]
+if_chain = "1.0"
```

Comment:
I would prefer that we try to keep our external dependencies as small as possible and this doesn't seem like a necessary addition to me but we have no official policy saying that, so I don't think it should hold up this PR.

Reply:
@wesleywiser if_chain is already in the source tree through clippy, this just adds it to rustc itself. I *could* rewrite it without if_chain, but it would be annoying and IMO a lot uglier.

Reply:
Yes, that's true but clippy isn't a dependency of rustc so any dependencies they pull in don't actually affect rustc. That's why you had to add it to the tidy allowed set of dependencies. 

It's my personal preference and we already have quite a lot of external dependencies as that list showed so don't worry about it 🙂